src/Router.tsx

```diff
@@ -13,23 +15,51 @@ const ImportImageWizard = lazy(
   () => import('./Components/CreateImageWizard/ImportImageWizard')
 );
 const CreateImageWizard = lazy(() => import('./Components/CreateImageWizard'));
+const isOnPremise = process.env.APP_TYPE_ENV;
+const useFlagWithAppTypeCheck = (
+  flagName: string,
+  flagHook: (flagName: string) => boolean
+) => {
+  // If it's the Cockpit app (on-premise), always return false
+  if (isOnPremise === 'cockpit') {
+    return false;
+  }
+
+  // Otherwise, return the actual flag value
+  return flagHook(flagName);
+};
```

Comment:
Can we move this fn out to `src/Utilities/useGetEnvironment` ? This will be useful everywhere we call out to useFlag.